nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -126,26 +133,9 @@ BtNavigator::on_configure(const rclcpp_lifecycle::State & /*state*/)
   auto node = shared_from_this();
   odom_smoother_ = std::make_shared<nav2_util::OdomSmoother>(node, 0.3, odom_topic_);
 
-  // Navigator defaults
-  const std::vector<std::string> default_navigator_ids = {
-    "navigate_to_pose",
-    "navigate_through_poses"
-  };
-  const std::vector<std::string> default_navigator_types = {
-    "nav2_bt_navigator/NavigateToPoseNavigator",
-    "nav2_bt_navigator/NavigateThroughPosesNavigator"
-  };
-
+  // Load navigator plugins
   std::vector<std::string> navigator_ids;
-  declare_parameter("navigators", default_navigator_ids);
   get_parameter("navigators", navigator_ids);
-  if (navigator_ids == default_navigator_ids) {
```

Comment:
This block is strictly required to declare the types of the defaults. This is non-optional. Please readd `default_navigator_types` definition and this block. You can use the `nav2_utils::declare_if_not_declared` function so that it'll only redeclare on lifecycle transition if not already declared. 

You could actually do that for `navigators` too so that you didn't have to change anything other than 2 lines. That would actually be closer in line with best practices.

Reply:
Using `declare_if_not_declared` was my first thought but it doesn't exist anymore.

Maybe default navigators are not necessary (yaml is always required) or I can also declare them in the constructor.

Reply:
I'm not sure what you mean by that, we have that in quite a few places in Nav2 :smile: 

https://github.com/ros-planning/navigation2/blob/main/nav2_navfn_planner/src/navfn_planner.cpp#L84

Its a `nav2_utils` function

Reply:
My bad, I was looking for `declare_if_not_declared` instead of `declare_parameter_if_not_declared` ...
I will make the modification.